pkg/tsdb/prometheus/prometheus.go

```diff
@@ -216,3 +217,16 @@ func parseResponse(value model.Value, query *PrometheusQuery) (*tsdb.QueryResult
 
 	return queryRes, nil
 }
+
+func IsAPIError(err error) bool {
+	var e *apiv1.Error
+	return errors.As(err, &e)
+}
+
+func WrapAPIError(err error) error {
+	var e *apiv1.Error
+	if errors.As(err, &e) {
+		return fmt.Errorf("%s: %s", e.Msg, e.Detail)
```

Comment:
What do you actually gain by this? `apiv1.Error` already implements the `error` interface, and `e.Error()` returns exactly the same string as here.